tekton/predicates_test.go

```diff
@@ -0,0 +1,145 @@
+package tekton_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	tekton "github.com/redhat-appstudio/integration-service/tekton"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	klog "k8s.io/klog/v2"
+
+	"knative.dev/pkg/apis"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+var _ = Describe("Predicates", func() {
+
+	const (
+		prefix                  = "testpipeline"
+		namespace               = "default"
+		PipelineTypeIntegration = "integration"
+	)
+	var (
+		testpipelineRun    *tektonv1beta1.PipelineRun
+		newtestpipelineRun *tektonv1beta1.PipelineRun
+	)
+	BeforeEach(func() {
+
+		testpipelineRun = &tektonv1beta1.PipelineRun{
+			ObjectMeta: v1.ObjectMeta{
+				GenerateName: prefix + "-",
+				Namespace:    namespace,
+			},
+			Spec: tektonv1beta1.PipelineRunSpec{
+				Params: []tektonv1beta1.Param{
+					{
+						Name: "output-image",
+						Value: tektonv1beta1.ArrayOrString{
+							StringVal: "test-image",
+						},
+					},
+				},
+			},
+			Status: tektonv1beta1.PipelineRunStatus{
```

Comment:
Maybe I'm wrong, but this should have no effect at all. You cannot set statuses this way.

Reply:
This got me thinking and I tried it out locally - the status does set in the `PipelineRun` object (hence why the unit tests are passing), but it doesn't actually get applied if we wanted to create that resource on the cluster.

Reply:
My bad. You are absolutely right. Most of our tests create objects so I didn't realize you don't do it here.